Fixing peer review issues

asmeurer opened the pull request:
See https://github.com/sympy/sympy-paper/issues/204. Feel free to push to this branch. Please make a note of which items are fixed in your commit messages so we can check them off. 

Here is the PeerJ guide to the rebuttal letter https://peerj.com/benefits/academic-rebuttal-letters/

asmeurer commented:
The rebuttal letter needs a cover letter. We should also copy and italicize all the reviewer comments into the letter.

asmeurer commented on paper.bib:
```diff
@@ -3577,7 +3577,7 @@ @incollection{dsl-little-languages
   chapter=3,
   pages={39-60},
-  editors={Peter H.\ Salas},
+  editor ={Peter H.\ Salas},
```
Oh derp.

asmeurer commented on rebuttal.tex:
```diff
@@ -0,0 +1,730 @@
+to a Float\dots''.  Numerical evaluation of the whole expression should
+always solve cancellation problem, as it was stated in the sentence
+``Applying the evalf method to the whole expression solves this problem.''
+\end{solution}
```
@fredrik-johansson can you comment on whether or not it is true that applying `evalf()` to the whole expression should _always_ avoid cancellation issues?

asmeurer replied:
If so, can you suggest how to improve the wording in the article, and if not, do you have a counterexample?

asmeurer commented:
@fredrik-johansson it would be helpful if you could review the changes to numerics.tex.

certik commented:
Working on the speed issue.

certik commented:
@asmeurer please review my wording in 4507c1591ac60a41bdb6b69571428aed26df0fdd.

certik commented:
@asmeurer I am done with the "speed" point. Please review my changes.

asmeurer commented:
Looks good. I added a small change to make it clear that SymEngine isn't used in 1.0 (which is the version used for the paper).

certik commented:
The acknowledgements were removed in 5ef75ae24816e907f2fe47927b3d7684dbf75b6e, which is not the right fix, so I reverted it in b0d4c326ab43ae1fcada6724eafa7a3bd0cde92c. Then I pushed in a proper fix, as the reviewer asked in b0d4c32...8441004 (this includes two unrelated @asmeurer's commits, but I don't know how to tell github not to include them).

@rpmuller please let me know if this change is ok with you. The reviewer asked:

> Lines 670--678 contain explanations what some of the authors' institutions are. It is customary to use acknowledgements to thank people and institutions, while institutions' details should be provided in the authors' footnotes in the documents' head.

So I did what he asked.

asmeurer commented:
@certik see https://github.com/sympy/sympy-paper/issues/205 regarding the acknowledgements.

certik commented:
@asmeurer looks like we still have to modify the last acknowledgement, but my change seems to be consistent with both the reviewer and #205. Let me know if you disagree.

asmeurer commented:
My understanding is that the acknowledgements section should be removed entirely.

rpmuller commented:
How can you have a paper without acknowledging the funding sources? You
certainly can't have LANL or SNL authors on it, if you choose to do so.

On Mon, Aug 22, 2016 at 4:42 PM Aaron Meurer notifications@github.com
wrote:

> My understanding is that the acknowledgements section should be removed
> entirely.
> 
> —
> You are receiving this because you were mentioned.
> 
> Reply to this email directly, view it on GitHub
> https://github.com/sympy/sympy-paper/pull/206#issuecomment-241573652,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AAchsrGIz0MdcrGk-K0Ym_dPCOSJVmlzks5qiiXHgaJpZM4JffXl
> .

asmeurer commented:
@rpmuller see https://peerj.com/about/author-instructions/cs#standard-sections. The funding information _is_ included, but we need to include it in the metadata on the PeerJ site when we (that is, I) submit, not in the article itself. It is then included in the final article by PeerJ. See for example, [this article](https://peerj.com/articles/cs-80/#additional-information).

rpmuller commented:
Got it. In that case, I'm fine with the changes. Sorry for the confusion.

On Mon, Aug 22, 2016 at 4:49 PM Rick Muller rpmuller@gmail.com wrote:

> How can you have a paper without acknowledging the funding sources? You
> certainly can't have LANL or SNL authors on it, if you choose to do so.
> 
> On Mon, Aug 22, 2016 at 4:42 PM Aaron Meurer notifications@github.com
> wrote:
> 
> > My understanding is that the acknowledgements section should be removed
> > entirely.
> > 
> > —
> > You are receiving this because you were mentioned.
> > 
> > Reply to this email directly, view it on GitHub
> > https://github.com/sympy/sympy-paper/pull/206#issuecomment-241573652,
> > or mute the thread
> > https://github.com/notifications/unsubscribe-auth/AAchsrGIz0MdcrGk-K0Ym_dPCOSJVmlzks5qiiXHgaJpZM4JffXl
> > .

certik commented:
@asmeurer, @rpmuller thanks for the clarifying it. I was actually confused too. Looks like that all is good, and it will be acknowledged just fine, I didn't know that PeerJ has a special section for this.

certik commented:
Regarding CASes, make sure you uncheck the box, since a reviewer asked to fix this.

asmeurer commented:
I did. I will likely end up sending an email to the editor about this, to see if the journal has a style guide that we should follow.

asmeurer commented:
Response from the editor on CASs vs. CAS's:

> This is one of the style points that we don't have a set policy on. We just ask that you be consistent throughout the article with whichever style you choose. I'm ccing your AE on this so he is in the loop on this particular point. 

So let's just use CAS's, as that was requested by the reviewer.

certik commented:
Didn't the reviewer request CASes?

Sent from my mobile phone.
On Aug 30, 2016 4:12 PM, "Aaron Meurer" notifications@github.com wrote:

> Response from the editor on CASs vs. CAS's:
> 
> This is one of the style points that we don't have a set policy on. We
> just ask that you be consistent throughout the article with whichever style
> you choose. I'm ccing your AE on this so he is in the loop on this
> particular point.
> 
> So let's just use CAS's, as that was requested by the reviewer.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/sympy/sympy-paper/pull/206#issuecomment-243564746,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AABQWLtzs2Pl6GEvtppa19rElm7UES5-ks5qlI7IgaJpZM4JffXl
> .

asmeurer commented on rebuttal.tex:
```diff
@@ -357,6 +357,21 @@ \section{Reviewer 3}
+      main paper content.  It either should stay in the supplement or we
+      should do a moderate cleanup for this section.  E.g. list only few
+      projects here and put other in the supplement.  Or/And show here
+      only project names.
```
I agree with this, but I think we'll need a stronger argument if we are going to leave it out.

asmeurer commented:
Does anyone know if it's possible to `\ref` supplement section numbers in the paper?

asmeurer commented:
I have added some rebuttal to the rewrite point. I would appreciate some feedback on it, though, as I'm sure it could be improved.

asmeurer committed:
Spelling

asmeurer committed:
Remove some of the "other projects that use SymPy"

asmeurer committed:
Be more forceful

asmeurer committed:
Remove sentence

asmeurer committed:
Remove a sentence

asmeurer committed:
Remove sentence

asmeurer committed:
Grammar

asmeurer committed:
Split something into an itemize

asmeurer committed:
Use the right quotes

asmeurer committed:
Itemize some text

asmeurer committed:
Spell out the submodule name

asmeurer committed:
Spell out section name

asmeurer committed:
Todos

asmeurer committed:
Make the items more parallel

asmeurer committed:
Grammar

asmeurer committed:
Reword

asmeurer committed:
Reword

asmeurer committed:
Some text

asmeurer committed:
Missing word

asmeurer committed:
Remove a sentence

asmeurer committed:
Some words

asmeurer committed:
Remove some words

asmeurer committed:
Change some words

asmeurer committed:
Some wording

asmeurer committed:
Add name of manuscript to letter

asmeurer committed:
Some more of the letter

asmeurer committed:
Delete sentence

asmeurer committed:
Add missing vertical space

asmeurer committed:
Fix some phrasing

asmeurer committed:
Phrasing

asmeurer committed:
Final paragraph for letter

asmeurer committed:
bigskip

asmeurer committed:
Move projects that depend on SymPy list to the main paper

asmeurer committed:
Remove the SymPy Live section from the supplement

We are just going to include it in the projects that use SymPy section. There
is not enough content here to warrant a separate supplement section.

asmeurer committed:
Add SymPy Live and SymPy Gamma to the projects that depend on SymPy section

asmeurer committed:
Rename other_projects_that_use_sympy.tex to projects_that_depend_on_sympy.tex

asmeurer committed:
Remove references to live.tex

asmeurer committed:
Fix order of files in tests and add missing files

asmeurer committed:
Fix failing Matrix doctest

asmeurer committed:
Move assumptions and basic usage into their own files

asmeurer committed:
Fix tests

We need to make sure that x, y, and z are never redefined with assumptions.

asmeurer committed:
Code quality

asmeurer committed:
Consistent non-usage of commas in symbols()

asmeurer committed:
Simplify some wording

Commit:
Fix broken references

asmeurer committed:
Add links for SymPy Live and Gamma

asmeurer committed:
Add citation for Cadabra

asmeurer committed:
Fix name and URL for OctSymPy

asmeurer committed:
Add an citation for OctSymPy

@cbm755 is this the right way to cite OctSymPy?

asmeurer committed:
Remove SageMathCloud from list of projects, since it somewhat duplicates SageMath

asmeurer committed:
Add several citations for the other projects

asmeurer committed:
galgebra description

asmeurer committed:
Add some more projects that use SymPy citations

asmeurer committed:
Convert other projects into a table

Still need to fill in the domain column

asmeurer committed:
\ref the table

asmeurer committed:
Change OctSymPy back to Octave Symbolic

As per @cbm755.

asmeurer committed:
Fix Octave stuff

asmeurer committed:
Sage -> SageMath

asmeurer committed:
Discuss Rete algorithm and new assumptions in the logic section

This should help our rebuttal point for keeping the supplement logic
subsection in the supplement.

Commit:
Proofreading

This fixes section numbers and references, that out of sync
after reordering of sections.

Commit:
Drop SymPy Live from table of projects, move to intro

* now we have the documentaion reference early
* save space
* we avoid some duplication

Commit:
Compactify projects table

Domain column was removed.  I believe it can't be very usefull.  E.g.
for most entries there should be something like "CAS".  Free text
in the Description should be enough.

Commit:
Compactify some symbol defs in Physics section

Commit:
Drop another assumption handler for gamma

We have one example, why second?

Commit:
Add automatic cross-references to the paper (in rebuttal)

See more at:
http://tex.stackexchange.com/questions/3060/referencing-labels-from-other-files

XXX: Section and footnotes numbers are correct, but my pdf viewer
(evince) doesn't open a new document if I click on them.  I hope,
this is a bug.

Commit:
Document in rebuttal.tex "other projects" section

asmeurer commented:
This is ready for people to start reviewing and signing off on.

asmeurer committed:
Remove acknowledgements section

This metadata should go in the funding statement on the PeerJ site, which will
then be part of the final document created by PeerJ. See
https://github.com/sympy/sympy-paper/issues/205 and
https://github.com/sympy/sympy-paper/pull/206#issuecomment-241577396.

fredrik-johansson commented on numerics.tex:
```diff
@@ -68,19 +79,30 @@ \subsection{The mpmath library}
+Like SymPy, mpmath is a pure Python library.  A design decision of SymPy is to
+keep it and its required dependencies pure Python. This is a primary advantage
+ of mpmath over other multiple precision libraries such as GNU MPFR~\cite{Fousse:2007:MMB:1236463.1236468},
+which is much more feature-rich and fast.  Like SymPy, mpmath is also BSD
```
I don't think MPFR is more feature-rich!

asmeurer replied:
@fredrik-johansson can you push a change to this branch for how you would rather word it (and note that this is in response to a comment by the reviewers, see the rebuttal document).

fredrik-johansson committed:
small change about mpfr

asmeurer committed:
Split the cover letter into a separate document so we can us the letter class

asmeurer committed:
Clean up the letter formatting a bit

asmeurer committed:
Add my signature

Uses method from http://texblog.org/2010/05/04/latex-letter-signature-as-image/

asmeurer committed:
Make my signature smaller

asmeurer committed:
Add the ERGS image to the cover letter

See https://tex.stackexchange.com/questions/42492/put-an-image-header-in-documentclassletter

certik committed:
Add `please`

certik committed:
Add the sentence back in

It was removed in 59dcf547dfd813fe91a377c6ddcfda6462f66dbf.

asmeurer committed:
Fix my office number

asmeurer committed:
Add the full text from reviewer 3

asmeurer committed:
Remove confusing sentence from letter

asmeurer committed:
Add pdf version of ERGS logo

asmeurer committed:
Capitalization

asmeurer committed:
Update logo-curves.pdf

Hopefully it has an alpha now.

asmeurer committed:
Fix the width of the projects that depend on SymPy table

asmeurer committed:
Fix the URL for Octave symbolic citation

asmeurer committed:
Fix the pages for the yt citation

asmeurer committed:
Remove redundant word

asmeurer committed:
Add citation for SymPy Gamma

asmeurer committed:
Note that it was moved from the supplement

asmeurer committed:
Note the section number

asmeurer committed:
Update mpfr rebuttal

asmeurer committed:
Move citations to the text in the projects table

asmeurer committed:
More consistent descriptions

asmeurer committed:
Fix some formatting

scopatz committed:
new performance

scopatz committed:
Merge branch 'peer-reviews' of github.com:sympy/sympy-paper into peer-reviews

asmeurer committed:
Don't use O

scopatz committed:
speed -> perf

scopatz committed:
Merge branch 'peer-reviews' of github.com:sympy/sympy-paper into peer-reviews

asmeurer committed:
Fix line wrapping

asmeurer committed:
Merge branch 'peer-reviews' of github.com:sympy/sympy-paper into peer-reviews

asmeurer committed:
Cross-reference the section

isuruf committed:
Fix typos in architecture

skirpichev committed:
Fix broken external links.

And order does matter.

asmeurer committed:
Remove "domain specific" from the abstract

"Domain specific" was renamed to "physics". Several submodules are discussed
in depth, so just say "submodules".

asmeurer committed:
Add a link to the issue

asmeurer committed:
Merge branch 'peer-reviews' of github.com:sympy/sympy-paper into peer-reviews

asmeurer committed:
Colorize the links in the rebuttal

scopatz committed:
minor paper

scopatz committed:
Merge branch 'peer-reviews' of github.com:sympy/sympy-paper into peer-reviews

asmeurer committed:
Set cross-document link color to black

asmeurer committed:
Add SciPy reference for LIL

asmeurer committed:
Fix minus sign

asmeurer committed:
Clear an unnecessary sentence

asmeurer committed:
Add a citation for Google App Engine

asmeurer committed:
Merge branch 'peer-reviews' of github.com:sympy/sympy-paper into peer-reviews

asmeurer committed:
Use the PDF logo

asmeurer commented:
@scopatz does the letterhead on the cover letter look good to you?

scopatz commented:
> @scopatz does the letterhead on the cover letter look good to you?

+1 from me!

asmeurer committed:
Modify SymPy Live paragraph

asmeurer committed:
Grammar

asmeurer committed:
Wording

asmeurer committed:
Fix reference number

asmeurer committed:
List the issue number

asmeurer committed:
More about SymPy Gamma in the rebuttal

asmeurer committed:
Fix some formatting

asmeurer committed:
Fix supplement ~ in front of \ref

asmeurer committed:
Rename \label{comp-mma} to \label{suppsec:comp-mma}

asmeurer committed:
Rename \label{other-proj} to \label{sec:other-proj}

asmeurer committed:
Add a section label for the basic usage section

asmeurer committed:
Add the supplement as an external document

asmeurer committed:
Fix Mathematica section \ref

asmeurer committed:
Name the section numbers in the reviewer 3 rewrite rebuttal

asmeurer committed:
Fix a \ref

asmeurer committed:
Clarify/correct the build instructions in the README

asmeurer committed:
Add how to run the tests to the README

asmeurer committed:
Rename SymPy Gamma label

asmeurer committed:
Fix supplement intro for SymPy Gamma

asmeurer committed:
Add a x-ref to the Gamma supplement section to the rebuttal

asmeurer committed:
Fix SymPy Live rebuttal

We removed the section entirely.

asmeurer committed:
Better word

asmeurer committed:
Change "in-depth" to "in depth" everywhere

Except for one place where it was in the reviewer's text.

asmeurer committed:
Punctuation

asmeurer committed:
Xref a footnote in the rebuttal

asmeurer committed:
Fix a typo

asmeurer committed:
Fix some reviewer numbering

asmeurer committed:
Grammar

asmeurer committed:
Better wording

asmeurer committed:
XRef the assumptions footnote to the rebuttal

asmeurer committed:
Missing word

asmeurer committed:
Logic is also mentioned in the extensibility section

asmeurer committed:
Note that we renamed the features section

asmeurer committed:
Replace repr quote in rebuttal with cross reference

asmeurer committed:
Cross reference the floating point section in the rebuttal

asmeurer commented on rebuttal.tex:
```diff
@@ -0,0 +1,948 @@
+   65536.0000000000
+   >>> ((phi**100-psi**100)/sqrt(5) - fibonacci(100)).evalf()
+   0.e-104
+\end{verbatim}
```
@fredrik-johansson how do you feel about the example currently in the paper (`cos(exp(-100)) - 1`) vs. this one? The paper states that this is a "notorious case". Is this expression a classic example, which can be cited from somewhere?

asmeurer committed:
Complete sentence

asmeurer committed:
Fix quotes and crossref module number

asmeurer committed:
Be consistent about "submodule" in the rebuttal

asmeurer committed:
Match the conclusion to the current paper sections

asmeurer committed:
Rearrange conclusion to match the current order of the paper

asmeurer committed:
Update point 39 of the rebuttal

asmeurer committed:
Clarify that the acknowledgments section was removed

asmeurer committed:
Cross reference gruntz footnote in rebuttal

asmeurer committed:
Use sympygamma.com instead of gamma.sympy.org

asmeurer committed:
Shouldn't use colon

asmeurer committed:
Add link to gamma page for tan(x)

asmeurer committed:
Add color to the links in the paper and supplement

What do people think? We can remove this if people don't like it. I wanted to
add it because some things that were links were subtle, and probably wouldn't
be noticed by most readers.

asmeurer commented:
Another thing we should do is check all the URLs in the documents to check that they are valid. Does anyone know of a tool that will check all URLs to make sure they don't 404?

asmeurer committed:
Better word

asmeurer committed:
Add sentence for projects that depend on SymPy to intro

asmeurer committed:
Wording

asmeurer committed:
"supplementary material" instead of "supplementary materials"

asmeurer committed:
Move immutability paragraph to the architecture section

asmeurer committed:
Reword

asmeurer committed:
Define "bedrock areas"

asmeurer committed:
Add sympy.calculus to the list of calculus submodules

asmeurer committed:
Add a citation for Xy-pic

asmeurer committed:
Use "integer" instead of "integral" (same word the assumptions system uses)

asmeurer committed:
Can we fix this overflowing text?

Maybe we should just leave it and let the journal figure it out.

asmeurer committed:
Grammar

asmeurer committed:
Parenthetical

asmeurer committed:
Fix the Makefile (simplification should be in the paper, not the supplement)

asmeurer committed:
Don't use longtable for the simplification table

It should only be used for tables that can't fit on one page.

asmeurer committed:
Fix simplification table width

asmeurer committed:
Use exp instead of E** in the limit example

asmeurer committed:
Wording

asmeurer committed:
Add an example for hyperexpand

Now all simplification functions from the table are represented.

asmeurer committed:
Reference simplification supplement

asmeurer committed:
The architecture section is now after the printers section

asmeurer committed:
Clarify a sentence

asmeurer committed:
Some examples, not more examples (no examples were given)

asmeurer committed:
Makefile: the supplement doesn't depend on solvers.tex

asmeurer committed:
Remove incorrect "and"

asmeurer committed:
Clarify matrices features list

I think the last items are all supposed to be related to computing inverses
(except for singular values, which was added later).

asmeurer committed:
Clarify that Dictionary of Keys uses a Python dictionary

asmeurer committed:
Cite the Rocklin PhD thesis

asmeurer committed:
Change "Subsection" to "section" for consistency

asmeurer committed:
"domain specific" to "domain specific physics"

asmeurer committed:
Clearer wording

asmeurer committed:
It should be Wigner D-matrix

asmeurer committed:
Clearer wording

asmeurer committed:
Remove ambiguity that could make it sound like Basic is part of Python

asmeurer committed:
Remove redundant word

asmeurer committed:
Definite article

asmeurer committed:
SymPy never uses the phrase "terminal node"

asmeurer committed:
Remove "so"

asmeurer committed:
"other examples" (one was just shown)

asmeurer committed:
Hashing is useful for any dict, not just caching

asmeurer committed:
The design isn't just simplified by interning

asmeurer committed:
Add another reference to the supplement

asmeurer committed:
Better wording

asmeurer committed:
e.g., not i.e.

asmeurer committed:
Better wording

asmeurer committed:
Grammar

asmeurer committed:
Better wording

asmeurer committed:
Better wording

asmeurer committed:
Better wording

asmeurer committed:
Sort the imports

asmeurer committed:
Fix syntax and unnecessary import in gamma example

asmeurer committed:
"fine-grained"

asmeurer committed:
I think these should be two sentences (previous wording did not make sense)

asmeurer committed:
Make it clear that SymEngine is not used in SymPy 1.0

asmeurer committed:
Fix wording

asmeurer committed:
"module" -> "package" (based on what we said in the rebuttal)

asmeurer committed:
The other ones say "Python package" so this one should too

skirpichev committed:
Fix overflow (assumptions)

asmeurer committed:
"This" doesn't make sense here

asmeurer committed:
Use a pronoun

asmeurer commented on paper.bib:
```diff
@@ -3590,3 +3598,149 @@ @misc{mpmath
+  title =     {{SciPy}: Open source scientific tools for {Python}},
+  year =      {2001--},
+  url = "http://www.scipy.org/",
+  note = {[Online; accessed 2016-09-28]}
```
Should we get rid of this (it shows up in the citation)?

asmeurer commented:
I've read through the paper and the rebuttal, and I'm +1 to them (although I still can't decide if we should just use "CASs", c.f. https://github.com/sympy/sympy-paper/commit/58453435aa7c377952461391e8c0776650bac625#commitcomment-19230766). Only question I have: should we use cross-document references to reference specific section numbers when we mention the supplement?

asmeurer committed:
Unnecessary comma

asmeurer committed:
Grammar, and use \eqref

asmeurer committed:
Grammar

asmeurer commented:
Why is the Brent and Zimmerman reference in the supplement (in the series section) rendering as `[Brent and Zimmerman]` instead of a number?

asmeurer committed:
Grammar

asmeurer committed:
Use \verb|^|

It looks a little bit better than \texttt{\^{}}, though still not great.

asmeurer committed:
List the unknown variables for completeness

asmeurer committed:
Explain the parametrized variables

asmeurer committed:
Break the solution up more evenly

It looks better this way.

asmeurer committed:
Wording

asmeurer committed:
Grammar

asmeurer committed:
Grammar fixes

asmeurer committed:
Fix typo

asmeurer committed:
Add an intro sentence for the examples supplement section

asmeurer committed:
Don't use first person

asmeurer committed:
Don't use the symbols('x:3') syntax

asmeurer committed:
Wording

asmeurer committed:
Don't mention the Wolfram|Alpha paid features

asmeurer committed:
Use MathJax instead of LaTeX for SymPy Gamma

asmeurer committed:
I disagree that ** for exponentiation is "unlike the usual programming convention"

Several languages use this, not just Python. It's also quite common even for
languages that don't have an exponentiation operator for ^ to mean xor, not
exponentiation.

asmeurer committed:
Grammar and wording

asmeurer commented:
I am also +1 to the supplement.

asmeurer committed:
Fix latex compilation

skirpichev committed:
Don't use comma for symbols()

skirpichev committed:
Fix Brent & Zimmermann bibentry

skirpichev committed:
I'm sorry, but you can't use verb use here.

Ondrej told me that "use" - means "depend on".  Please be consistent
and don't break your own rules.

skirpichev committed:
This fixes for me broken refs in pdf index

(to the "references" sections)

skirpichev committed:
or -> "overloadability of the functionality" ?

skirpichev committed:
I doubt that ^ as power op is "more common".

Lets just state the fact, without subjective judgements.

asmeurer committed:
Rename image files with figure numbers

This was requested by the PeerJ staff. See
https://github.com/sympy/sympy-paper/issues/205.

asmeurer commented:
Thank you to everyone who helped with the reviews. I am going to resubmit the paper now. Please let me know right away if there are any objections.
